From 5dff10d6a3c2e19eed46af4639ade2ff0ca10d98 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sat, 18 Jul 2020 02:22:45 +0200
Subject: [PATCH] Guard glPointSize with GL_ES_VERSION_3_0

glPointSize is not available in GLES, so just disable the call on GLES
like done elsewhere.
---
 Rendering/ContextOpenGL2/vtkOpenGLContextDevice2D.cxx | 2 ++
 Rendering/ContextOpenGL2/vtkOpenGLContextDevice3D.cxx | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/Rendering/ContextOpenGL2/vtkOpenGLContextDevice2D.cxx b/Rendering/ContextOpenGL2/vtkOpenGLContextDevice2D.cxx
index cc376934dd..5f4eed5401 100644
--- a/Rendering/ContextOpenGL2/vtkOpenGLContextDevice2D.cxx
+++ b/Rendering/ContextOpenGL2/vtkOpenGLContextDevice2D.cxx
@@ -1912,7 +1912,9 @@ void vtkOpenGLContextDevice2D::SetPointSize(float size)
   {
     gl2ps->SetPointSize(size);
   }
+#ifndef GL_ES_VERSION_3_0
   glPointSize(size);
+#endif
 }
 
 //-----------------------------------------------------------------------------
diff --git a/Rendering/ContextOpenGL2/vtkOpenGLContextDevice3D.cxx b/Rendering/ContextOpenGL2/vtkOpenGLContextDevice3D.cxx
index d24eeeda75..ab46faf8ca 100644
--- a/Rendering/ContextOpenGL2/vtkOpenGLContextDevice3D.cxx
+++ b/Rendering/ContextOpenGL2/vtkOpenGLContextDevice3D.cxx
@@ -452,7 +452,9 @@ void vtkOpenGLContextDevice3D::DrawPoints(
 
   this->EnableDepthBuffer();
 
+#ifndef GL_ES_VERSION_3_0
   glPointSize(this->Pen->GetWidth());
+#endif
 
   vtkOpenGLHelper* cbo = nullptr;
   if (colors)
-- 
2.27.0

